### ISSUE TITLE
Not able to render kustomization file using helmCharts with oci:// repository url.

### ISSUE THREAD

**Issue by JanGrosse** [akuity/kargo-render#222]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

We are using kustomize as the main templating language in our repositories. Sometimes we combine this with `helmCharts`.
When I want to render such a kustomization template to plain files I receive an error if the kustomization uses `helmCharts` and specifies a `oci://` repository.

# Steps to Reproduce

Create a kustomization.yaml which uses a `helmChart` in your repository inside a `kargo` directory e.g.
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

helmCharts:
- name: kargo 
  version: 0.2.0
  repo: oci://ghcr.io/akuity/kargo-charts/kargo
  valuesFile: values.yaml
  namespace: kargo 
  releaseName: kargo 
  includeCRDs: true
```

Add a kargo-render.yaml
```
configVersion: v1alpha1
branchConfigs:
- pattern: test
  appConfigs:
    kargo:
      configManagement:
        kustomize:
          path: kargo
          enableHelm: true
      outputPath: kargo
```

Run local docker 
```
docker run -it ghcr.io/akuity/kargo-render:v0.1.0-rc.33 kargo-render render  --repo "<repository>"  --repo-username "<user>"   --repo-password  "<password>" --target-branch "test"
```

Receive error
```
Error: error pre-rendering manifests: error generating manifests using Argo CD repo server: `kustomize build /tmp/3315312386/repo/kargo --enable-helm --load-restrictor LoadRestrictionsRootOnly` failed exit status 1: Error: Error: looks like "oci://ghcr.io/akuity/kargo-charts/kargo" is not a valid chart repository or cannot be reached: object required
: unable to run: 'helm pull --untar --untardir /tmp/3315312386/repo/kargo/charts --repo oci://ghcr.io/akuity/kargo-charts/kargo kargo --version 0.2.0' with env=[HELM_CONFIG_HOME=/tmp/kustomize-helm-4020257331/helm HELM_CACHE_HOME=/tmp/kustomize-helm-4020257331/helm/.cache HELM_DATA_HOME=/tmp/kustomize-helm-4020257331/helm/.data] (is 'helm' installed?): exit status 1
```
# Screenshots
Also tested with kargo v0.2.0 results in the same error as local

![kargo-render-error](https://github.com/akuity/kargo-render/assets/25822756/6d5ab238-1c2a-4279-836b-7f5030913676)

# Version

Tested with `ghcr.io/akuity/kargo-render:v0.1.0-rc.33`. 
I wanted to test with the version from the docs (`ghcr.io/akuity/kargo-render:v0.1.0-rc.25`) but this seems no longer available. It results just in
```
Unable to find image 'ghcr.io/akuity/kargo-render:v0.1.0-rc.25' locally
docker: Error response from daemon: manifest unknown.
See 'docker run --help'.
```

# Additional Notes

I think this is related to [this issue](https://github.com/kubernetes-sigs/kustomize/issues/4381) which got fixed in kustomize with this [PR](https://github.com/kubernetes-sigs/kustomize/pull/5167)
ArgoCD also bumped the kustomize version to 5.2.1 with the [2.9.0 Release](https://github.com/argoproj/argo-cd/pull/16054#issuecomment-1772815169).

I checked the go.mod file and saw tha kargo-render still uses ArgoCD in version [2.8.4](https://github.com/akuity/kargo-render/blob/bb12522c1582bd1c367e6d2ceb292b76d4f182a8/go.mod#L62) - if this is already the issue then the [open PR](https://github.com/akuity/kargo-render/pull/221) would fix my issue already :-)

Greetings,
Jan.

**Comment by krancour** [comment]
> I checked the go.mod file and saw tha kargo-render still uses ArgoCD in version [2.8.4](https://github.com/akuity/kargo-render/blob/bb12522c1582bd1c367e6d2ceb292b76d4f182a8/go.mod#L62) - if this is already the issue then the https://github.com/akuity/kargo-render/pull/221 would fix my issue already :-)

I love it when it's that simple!

I will forewarn you, however, that `main` has significant breaking changes to the schema for the `kargo-render.yaml` file, so while an upcoming rc.34 may fix your issue, you will also have to update your config.

We will, of course, document those changes. Just didn't want you caught by surprise if you happened to build from source.

**Comment by krancour** [comment]
Fixed by #221
